scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -138,7 +144,8 @@ def nan_policy_1d(hypotest, data1d, unpacker, *args, n_outputs=2,
             if np.any(np.isnan(sample)):
                 raise ValueError("The input contains nan values")
 
-    elif nan_policy == 'propagate':
+    elif (nan_policy == 'propagate'
+          and hypotest not in override_propagate_funcs):
```

Comment:
Note that `stats.mode not in override_propagate_funcs` is `False`, so none if the `if`/`elif` special cases apply, and the test relies on the behavior of `mode` for 1D slice to get the reference value (just like it would if there were no NaNs at all).